### PR TITLE
fix(41212,41334): removed duplicate interface extensions

### PR DESCRIPTION
**Pull request by neotan**
Fixes [#41212](https://github.com/microsoft/TypeScript/pull/41212) and [#41334](https://github.com/microsoft/TypeScript/issues/41334)

### CODE REVIEW

**Review by saschanaz — CHANGES_REQUESTED** [review]
Oh, the real fix would be removing such duplicates from files like `addedTypes.jsonc`. Please remove `HTMLElement` and `SVGElement` from there instead.

But `uniq` seems still helpful with a different form, how about `assertUnique` that throws when there is a duplicate?

File: src/build/helpers.ts [hunk 1]
```diff
@@ -325,3 +325,8 @@ export function followTypeReferences(
     }
   }
 }
+
+export function uniq(list: string[] = []): string[] {
```
Comment (saschanaz) [inline src/build/helpers.ts:329]: `=[]` seems redundant when all callers pass a valid array.

File: src/build/helpers.ts [hunk 2]
```diff
@@ -325,3 +325,21 @@ export function followTypeReferences(
     }
   }
 }
+
+export function assertUnique(list: string[]): string[] {
+  if (!Array.isArray(list)) return list;
```
Comment (saschanaz) [inline src/build/helpers.ts:330]:
```suggestion
```

The type requirement `: string[]` is sufficient here (as this is just an internal function and thus is not exposed as a part of JavaScript library)

File: src/build/helpers.ts (hunk 1 above)
Comment (neotan) [reply src/build/helpers.ts]: Hi @saschanaz, 
Thanks for your comments! I've just started using TS these months, this is my first PR for TS ecosystem XD actually :P.
I updated the codes based on your comment, for the `assertUnique`, didn't use while/for/map for higher performance, but it's easier to understand.

Pls let me know if anything needs to be done. Thanks ;)

File: src/build/helpers.ts [hunk 3]
```diff
@@ -325,3 +325,21 @@ export function followTypeReferences(
...
+    .forEach((str, _, trimmed) => {
+      if (trimmed.indexOf(str) !== trimmed.lastIndexOf(str)) result.add(str);
+    });
+
+  if (result.size) {
+    throw new Error(
+      `Duplicate items [${[...result].join(", ")}] found in the list.`
+    );
+  }
+  return list;
```
Comment (saschanaz) [inline src/build/helpers.ts:344]: Good but IMO a bit overkill, checking whether it has a duplicate is enough IMO. A human should easily see the duplications in the list, given that callers are only expected to pass small lists.

```suggestion
  if (new Set(list).size < list.length) {
    throw new Error(`Duplicate items found in the list: ${list}`);
  }
  return list;
```

**Comment by saschanaz** [comment]
Thank you, looks good 👍

@orta I think I'm often blocked by "First-time contributors need a maintainer to approve running workflows" thing, what do you think about https://github.com/OSS-Docs-Tools/code-owner-self-merge/issues/26 ?

**Comment by orta** [comment]
( I agree. I can build that this week, I was thinking about giving access to closing PRs and maybe issues too 👍🏻 )

**Comment by saschanaz** [comment]
> 
> 
> ( I agree. I can build that this week, I was thinking about giving access to closing PRs and maybe issues too 👍🏻 )
> 
> `<edit>`and then I saw you had already got that ball rolling - thanks

That's https://github.com/OSS-Docs-Tools/code-owner-self-merge/pull/25

**Comment by saschanaz** [comment]
LGTM

**Pull request merged**